### ISSUE TITLE
Exception when running LSIF tool with npm rewriter

### ISSUE THREAD

**Issue by dbaeumer** [microsoft/lsif-node#20]
```
{"id":1,"type":"vertex","label":"metaData","version":"0.1.0"}
C:\src\VSCloudKernel\bin\LSIFLanguageServer\Debug\netcoreapp2.1\node_modules\npm-lsif\lib\main.js:89
        this.outDir = outDir.charAt(outDir.length - 1) !== '/' ? outDir + '/' : outDir;
                             ^

TypeError: Cannot read property 'charAt' of undefined
    at ExportLinker.addOutAndRoot (C:\src\VSCloudKernel\bin\LSIFLanguageServer\Debug\netcoreapp2.1\node_modules\npm-lsif\lib\main.js:89:30)
    at Interface.rd.on (C:\src\VSCloudKernel\bin\LSIFLanguageServer\Debug\netcoreapp2.1\node_modules\npm-lsif\lib\main.js:278:34)
    at emitOne (events.js:116:13)
    at Interface.emit (events.js:211:7)
    at Interface._onLine (readline.js:280:10)
    at Interface._normalWrite (readline.js:422:12)
    at Socket.ondata (readline.js:139:10)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
events.js:183
      throw er; // Unhandled 'error' event


```

**Comment by dbaeumer** [comment]
As suggested by @homitc workaround is to add a `root` and `out` property. However for JS project that feels wrong and we should be able to infer this from the include property.

@BertanAygun

**Comment by jepetty** [comment]
@dbaeumer is this still an open issue? We'd like to get this fixed for a scale investigation we're doing (makes it really challenging to automate typescript/javascript if we have to add settings 😉 ). If not, happy to self-assign and take a stab at it (of course, with your guidance on any unexpected behavior I try to introduce)

**Comment by dbaeumer** [comment]
Will look into this but I have no plans to backport :-). So it will be available on v0.4.x. Is this OK?

**Comment by jepetty** [comment]
Yup, that's perfect! Thanks for the help! :)

**Comment by dbaeumer** [comment]
Fixed in 0.4.2.

Fix was actually relatively easy. So if you need a back port let me know.
